### PR TITLE
Register account page created

### LINKED ISSUE #21: Register Account Page
This is the proper register account page. This page will contain the components created in the previous task and manage them.
Keep track of the the components to show based on step completed.
At the end, for create account button, call the api endpoint to create a user and pass in all the proper fields

### PR DESCRIPTION
**Pull request by RedPenguin88**
# Overview

This PR completes the registration page and thus closes #21. In essence, this PR combines the three components made in these PRs:
- #24
- #40 
- #44

and implements them into a single registration page.

## Notable Changes
- Adds new register page (Can be accessed by appending '/register' to the url, but can be easily accessed through the hyperlink in the login page)
  - Using react router, the buttons on the register page should behave as expected.
  - In terms of feedback, recommendations for best practices and the like are very welcome, since I am largely unfamiliar with the technologies here.
- Moves the pages made in previous PRs to the components directory for use with the register page.
- Makes significant modifications to the GeneralInfo, SignUp, and CreateAccount components for consistent styling.

## Usage
The register page can easily be accessed through the new 'register here' hyperlink on the login page.
register
From there, they are directed towards the General Info page created in #40. The buttons are now functional, bringing the user to each step of the account creation process. Each of the three components were modified to bring them into a consistent style, including making sure those components use our custom components like CustomButton.

After the user finishes inputting every field, the create account button will simply console log every field. Guidance on how to call the api endpoint to create the user would be much appreciated.

### Potential Updates
These are updates which would be suitable for future PRs.

- [ ] Right now, clicking the back button correctly returns to the previous page, however, it does not retain the form data the user had previously input, instead resetting each field to an empty string. An update should be made to retain this information when the "back" button is pressed.
- [ ] Right now, the user can continue to the next page without inputting anything into the text fields. Obviously most of the fields (if not all) are required. A modification should be made to ensure that the user inputs in all required fields before continuing.

### CODE REVIEW

**Comment by RedPenguin88** [comment]
Requesting review from @tbmcallister and @B-Timok as I made changes to files they worked on.

**Review by B-Timok — APPROVED** [review]
Everything looks good on my end. Nice job making the design more consistent. I'll look into the issue of making the input fields *required*

File: client/src/components/GeneralInfo/GeneralInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            // Store all form data
+            updateData('firstName', firstName);
+            updateData('lastName', lastName);
+            updateData('userName', userName);
+            updateData('email', email);
+            updateData('phone', phone);
+            updateData('DoB', DoB);
+            handleSubmit();
+          }}
+        >
```
Comment (Aligary) [inline client/src/components/GeneralInfo/GeneralInfo.tsx:51]: Have this onClick function be created outside of the return statement and only call it here. This way we only see the component and not the logic of the function.

File: client/src/components/SignUp/SignUp.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        <TextField header="Zip Code" setInputValue={setZipValue} type="zip" />
+        <CustomButton
+          onClick={() => {
+            updateData('address', address);
+            updateData('city', city);
+            updateData('state', state);
+            updateData('zip', zip);
+            handleSubmit();
+          }}
+        >
```
Comment (Aligary) [inline client/src/components/SignUp/SignUp.tsx:54]: same here

**Review by Aligary — APPROVED** [review]
Thanks for refactoring these pages into components. I left some minor comments but they aren't critical and should okay!

**Review by trevortippery — APPROVED** [review]
Looks good on my end. No complaints. Approved 🎉

**Comment by RedPenguin88** [comment]
10 commits behind! Please forgive me if I blunder the merging process.

File: client/src/components/CreateAccount/CreateAccount.tsx [hunk 3]
```diff
@@ -37,7 +38,15 @@ export default function CreateAccount(): JSX.Element {
       <div className={styles['CreateAccount__PaddedDiv']}>
         <PasswordField header="Password" setInputValue={setPassword} />
         <PasswordField header="Confirm Password" setInputValue={setConfirmPassword} />
-        <CustomButton onClick={handleClick}>Create Account</CustomButton>
+        <CustomButton
```
Comment (morenj) [inline client/src/components/CreateAccount/CreateAccount.tsx:41]: Great use of the customButton component!

**Review by morenj — APPROVED** [review]
Looks good on my end, great job. Thank you for your continuous contributions.

**Comment by RedPenguin88** [comment]
Please refer to #89. I was unable to fix the merge conflicts for this PR.

**Pull request closed without merging**